steam/ext/dota2/models.py

```diff
@@ -0,0 +1,490 @@
+
+    async def post_social_message(self, message: str) -> None:
+        """Post message in social feed.
+
+        Currently, messages sent with this are visible in "User Feed - Widget" of Profile Showcase.
+        This functionality was possible long ago naturally in the Dota 2 client.
+        """
+        await self._state.post_social_message(message=message)
+
+
+class ProfileCard:
+    def __init__(self, proto: common.ProfileCard):
+        self.account_id = proto.account_id
+        self.badge_points = proto.badge_points
+        self.event_points = proto.event_points
+        self.event_id = proto.event_id
+        self.recent_battle_cup_victory = proto.recent_battle_cup_victory
+        self.rank_tier = RankTier.try_value(proto.rank_tier)
+        """Ranked medal like Herald-Immortal with a number of stars, i.e. Legend 5."""
+        self.leaderboard_rank = proto.leaderboard_rank
+        """Leaderboard rank, i.e. found here https://www.dota2.com/leaderboards/#europe."""
+        self.is_plus_subscriber = proto.is_plus_subscriber
+        """Is Dota Plus Subscriber."""
+        self.plus_original_start_date = proto.plus_original_start_date
+        """When user subscribed to Dota Plus for their very first time."""
+        self.favorite_team_packed = proto.favorite_team_packed
+        self.lifetime_games = proto.lifetime_games
+        """Amount of lifetime games, includes Turbo games as well."""
+
+        # (?) Unused/Deprecated by Valve
+        # self.slots = proto.slots  # profile page was reworked
+        # self.title = proto.title
+        # self.rank_tier_score = proto.rank_tier_score  # relic from time when support/core MMR were separated
+        # self.leaderboard_rank_core = proto.leaderboard_rank_core  # relic from time when support/core MMR were separated
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} account_id={self.account_id}>"
+
+
+class LiveMatchPlayer(PartialUser):
+    def __init__(
+        self,
+        state: GCState,
+        id: Intable,
+        hero: Hero,
+        team: int,
+        team_slot: int,
+    ) -> None:
+        super().__init__(state, id)
+        self.hero = hero
+        self.team = team
+        self.team_slot = team_slot
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} id={self.id} hero={self.hero!r}>"
+
+
+class MatchMinimalPlayer:
+    def __init__(self, state: GCState, proto: common.MatchMinimalPlayer):
+        self._state = state
+
+        self.id = proto.account_id
+        self.hero = Hero.try_value(proto.hero_id)
+        self.kills = proto.kills
+        self.deaths = proto.deaths
+        self.assists = proto.assists
+        self.items = proto.items
+        self.player_slot = proto.player_slot
+        self.pro_name = proto.pro_name
+        self.level = proto.level
+        self.team_number = proto.team_number
+
+
+class MatchMinimal:
```

Comment:
Might be useful to have all of these subclass a common base that sets the basic attributes a match is expected to have that is a subclass of PartialMatch